manager/manager.go

```diff
@@ -0,0 +1,295 @@
+package manager
```

Comment:
Lets add package level docs here
```suggestion
// package manager contains ....
package manager
```

These show up in a special way in the generated docs. Also we should write things like
this one - https://github.com/flyteorg/flyteadmin/blob/master/scheduler/doc.go

Reply:
I want to understand the assumptions, the high level design and what are unhandled cases

Reply:
done